### PR TITLE
Add terraform package

### PR DESCRIPTION
**Pull request by nsmith5**
Terraform!

### CODE REVIEW

**Review by imjasonh — DISMISSED** [review]

**Review by luhring — DISMISSED** [review]

File: terraform.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - go
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/hashicorp/terraform
+      branch: v${{package.version}}
+      destination: terraform
```
Comment (tuananh) [inline terraform.yaml:25]: use tag instead of branch and `expected-commit`?

File: terraform.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  - runs: |
+      cd terraform
+      go generate ./...
+      CGO_ENABLED=0 go build -v
+  - runs: |
+      install -m755 -D ./terraform/terraform "${{targets.destdir}}/usr/sbin/terraform"
```
Comment (tuananh) [inline terraform.yaml:31]: why `sbin`?

File: terraform.yaml (hunk 1 above)
Comment (nsmith5) [reply terraform.yaml:24]: Good idea!

File: terraform.yaml (hunk 2 above)
Comment (nsmith5) [reply terraform.yaml]: Just some copy pasta from your Vault example. bin makes sense

**Review by imjasonh — APPROVED** [review]

File: terraform.yaml (hunk 2 above)
Comment (tuananh) [reply terraform.yaml]: olps :D

**Pull request merged**